datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -728,6 +729,30 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
                 },
             },
 
+            // unfold the expression round(source, n)
+            Expr::ScalarFunction {
+                fun: BuiltinScalarFunction::Round,
+                args,
+            } => match &args[..] {
+                [source, n] => {
+                    println!("I am here, source = {:?}, n = {:?}", source, n);
+                    let exp = Box::new(power(
+                        lit(10.0),
+                        Expr::Cast(Cast::new(n.clone().into(), DataType::Float64)),
```

Comment:
One problem here is that we will always convert the data type to `Float64` to adapt the `power` function